2048/2048/src/main/java/com/appcoins/eskills2048/api/GeneralPlayerStats.java

```diff
@@ -11,6 +11,7 @@
   @Headers("accept: application/json") @GET("room/statistics/general_player_stats")
   Single<GeneralPlayerStatsResponse> getGeneralPlayerStats(
       @Query("package_name") String packageName, @Query("wallet_address") String walletAddress,
-      @Query("ranks_above_user") int ranksAbove, @Query("ranks_below_user") int ranksBelow,
-      @Query("time_frame") String timeframe);
+      @Query("product") String product, @Query("ranks_above_user") int ranksAbove,
+      @Query("ranks_below_user") int ranksBelow,
+      @Query("match_environment") String matchEnvironment, @Query("time_frame") String timeframe);
```

Comment:
you don't need to convert this to String. you can just use Environment type and let retrofit deal with the conversion (see SetFinalScoreUseCase)

Reply:
fixed